auditing/auditing-interceptor.go

```diff
@@ -25,9 +25,10 @@ const (
 	Exclude string = "exclude-from-auditing"
 )
 
-func UnaryServerInterceptor(a Auditing, logger *zap.SugaredLogger, shouldAudit func(fullMethod string) bool) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(a Auditing, logger *slog.Logger, shouldAudit func(fullMethod string) bool) grpc.UnaryServerInterceptor {
```

Comment:
Would be great to change the signature so the function returns `(grpc.UnaryServerInterceptor, error)` so the caller has to handle the case if the `a` parameter is nil.

Reply:
done